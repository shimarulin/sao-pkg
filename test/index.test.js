const path = require('path')
const sao = require('sao')

const generator = path.join(__dirname, '..')

const getPkg = (pkg, fields) => {
  pkg = JSON.parse(pkg)
  return fields.reduce((res, curr) => {
    res[curr] = pkg[curr]
    return res
  }, {})
}

test('use defaults', async () => {
  const helper = await sao.mock({ generator })
  expect(helper.fileList).toMatchSnapshot()
  expect(
    getPkg(await helper.readFile(
      'package.json'),
      ['name', 'author'])
  ).toMatchSnapshot('package.json')
})
