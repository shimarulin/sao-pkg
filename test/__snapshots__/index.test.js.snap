// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file list with use defaults 1`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  "LICENSE",
  "README.md",
]
`;
